src/content/posts/whats-possible-with-melange.mdx

```diff
@@ -0,0 +1,202 @@
+[Make sure your code is universal](https://ml-in-barcelona.github.io/server-reason-react/local/server-reason-react/universal-code.html).
+
+![Example of a react component rendered with both reason-react and server-reason-react](/images/server-reason-react-graph.png)
+
+## Preprocessing via ppx becomes better and faster
+
+ReScript doesn’t recommend the usage of ppx's and has been looking to replace them since the split. His claims are valid: they are a blackbox, are limited, slows the compilation and are hard to implement right.
+
+On the Melange side, the approach is the oppositve. Empower ppx’s via [ppxlib](https://github.com/ocaml-ppx/ppxlib) and dune, which they integrate smoothly. [ppxlib](https://github.com/ocaml-ppx/ppxlib) is the framework to create high quality preprocessors, it also integrates perfectly with dune via `(preprocess ...)` and also:
+
+- **Ppx runs faster**. No need to serialise/deserialise the AST on each ppx call, neither run a binary.
+- Ppx are smarter. Dune has a few methods to make ppx run in a single pass, and ppxlib allows to create context free transformations. Making sure rewrites don’t do unnecessary work.
+- No need to distribute pre-built binaries. ppx's are dune libraries, and no need to have GitHub actions to create per-distribution artifacts.
+- This gives us compatibility with more distributions such as M1, M2, any linux distro and even Windows.
+- Access to all [ppx's from opam](https://ocaml.org/packages/search?q=ppx_), such as [ppx_enumerate](https://ocaml.org/p/ppx_enumerate/latest), [ppx_log](https://ocaml.org/p/ppx_log/latest), [ppx_expect](https://ocaml.org/p/ppx_expect/latest), [ppx_inline_test](https://ocaml.org/p/ppx_inline_test/latest), [ppx_blob](https://ocaml.org/p/ppx_blob/latest), [ppx_fields_conv](https://ocaml.org/p/ppx_fields_conv/latest), [ppx_yojson_conv](https://ocaml.org/p/ppx_yojson_conv/latest), [ppx_deriving_rpc](https://ocaml.org/p/ppx_deriving_rpc/latest).
+<sup>Some of this ppx's still need to support melange, and aren't available out of the box</sup>
+
+## Different configs for development and production
+
+Seems basic to showcase this, but previously in BuckleScript you can’t setup different config for production and development. That’s not the case for dune, where you are able to have different [profiles]() and set everything you need differently. Warnings, instrumentation and some libraries are the most common to load differently.
+
+Previously we had a script that generates the config by certain parameter and accomplish such a basic functionality.
+
+## Integration with documentation tooling (odoc)
+
+The other benefit of integrating with the OCaml platform is the documentation tooling is available. It’s called [**odoc**](https://github.com/ocaml/odoc) and it will generate a documentation website from your interface files, with navigation, cross-linking and many necessities
```

Comment:
```suggestion
The other benefit of integrating with the OCaml platform is the documentation tooling is available. It’s called [**odoc**](https://github.com/ocaml/odoc) and it will generate a documentation website from your interface files, with navigation, cross-linking and other utilities.
```